src/read/pe/file.rs

```diff
@@ -44,13 +59,15 @@ where
     pub fn parse(data: R) -> Result<Self> {
         let dos_header = pe::ImageDosHeader::parse(data)?;
         let mut offset = dos_header.nt_headers_offset().into();
+        let rich_header_infos = RichHeaderInfos::parse(data, offset);
```

Comment:
So that would be turned into
```rust
impl PeFile {
    // ...
    pub fn rich_header_infos(&self) -> Option<&'data RichHeaderInfos> {
        RichHeaderInfos::parse(self.data, dos_header.nt_headers_offset());
    }
}
```

Should I cache the result somewhere? Is it OK to keep this cached result in the struct PeFile (as a `Option<&'data RichHeaderInfos>` field?)

Reply:
The goal is that users should only pay for what they need. So it would be preferable for the user to cache this themselves if needed. I expect that the user will already be storing other data as they parse the file so I don't see this as a burden.